R/Organism.R

```diff
@@ -878,12 +880,15 @@ setMethod("simBac", "Bac", function(object, arena, j, sublb, bacnum, sec_obj="no
   }
   pos <- arena@orgdat[,c('x','y')]
   if(!dead && !arena@stir && object@speed != 0){
-    if(object@chem == ''){
+    if(is.null(object@chem)){
       mov_pos <- move(object, pos, arena@n, arena@m, j, arena@occupyM)
       arena@orgdat[,c('x','y')] <- mov_pos
     }else{
-      chemo_pos <- chemotaxis(object, arena, j)
+      for (v in seq_along(object@chem)){
```

Comment:
Read the vector and apply the chemotaxis function for each compound by the given order